types/address.go

```diff
@@ -256,6 +264,21 @@ func (aa AccAddress) Format(s fmt.State, verb rune) {
 	}
 }
 
+// LengthPrefixAddress prefixes the address bytes with its length, this is used
+// for variable-length components in store keys. Note: All addresses should be
+// max 255 bytes, or else this function panics.
+func LengthPrefixAddress(bz []byte) []byte {
+	if len(bz) == 0 {
+		return bz
```

Comment:
shouldn't we return `{0}` instead of empty bytes?

Reply:
The question is: is `{0}` a valid address? I would say no. ref: https://github.com/cosmos/cosmos-sdk/pull/8363#discussion_r564025693

Reply:
Not exactly. This function (at least it in it's description and definition) doesn't say anything if it validates an address and if it's responsible for handling zero addresses: what is an invalid address and how do we handle it?

I see that you changed the function by adding `error` - this is much better and I think it solves this issue. I will add more comments in new threads.